concordium-consensus/src/Concordium/Scheduler/WasmIntegration/V1.hs

```diff
@@ -0,0 +1,549 @@
+        initCtxBytes = encodeChainMeta cm <> encodeReceiveContext receiveCtx
+        amountWord = _amount amnt
+        stateBytes = contractState cs
+        energy = fromIntegral initialEnergy
+        paramBytes = BSS.fromShort (parameter param)
+        nameBytes = Text.encodeUtf8 (receiveName rName)
+
+-- |Resume execution after processing the interrupt. This can only be called once on a single 'ReceiveInterruptedState'.
+resumeReceiveFun ::
+    ReceiveInterruptedState
+    -> Maybe ContractState -- ^State of the contract to start in.
+    -> Amount -- ^Current balance of the contract, if it changed.
+    -> InvokeResponseCode
+    -> Maybe ReturnValue
+    -> InterpreterEnergy  -- ^Amount of energy available for execution.
+    -> Maybe (Either ContractExecutionReject ReceiveResultData, InterpreterEnergy)
+    -- ^Nothing if execution used up all the energy, and otherwise the result
+    -- of execution with the amount of energy remaining.
+resumeReceiveFun is cs amnt statusCode rVal remainingEnergy = unsafePerformIO $ do
+              withReceiveInterruptedState is $ \isPtr ->
+                withStateBytes $ \(stateBytesPtr, stateBytesLen) ->
+                  withMaybeReturnValue rVal $ \rValPtr ->
+                    alloca $ \outputLenPtr -> alloca $ \outputReturnValuePtrPtr -> do
+                      outPtr <- resume_receive isPtr
+                                              newStateTag
+                                              (castPtr stateBytesPtr) (fromIntegral stateBytesLen)
+                                              amountWord
+                                              (invokeResponseToWord64 statusCode)
+                                              rValPtr
+                                              energy
+                                              outputReturnValuePtrPtr
+                                              outputLenPtr
+                      if outPtr == nullPtr then return (Just (Left Trap, 0)) -- this case should not happen
+                      else do
+                        len <- peek outputLenPtr
+                        bs <- BSU.unsafePackCStringFinalizer outPtr (fromIntegral len) (rs_free_array_len outPtr (fromIntegral len))
+                        returnValuePtr <- peek outputReturnValuePtrPtr
+                        processReceiveResult bs returnValuePtr (Left is)
+    where
+        (withStateBytes, newStateTag) = case cs of
+           Just stateBytes -> (BSU.unsafeUseAsCStringLen (contractState stateBytes), 1::Word8)
+           Nothing -> (\f -> f (nullPtr, 0), 0::Word8)
+        energy = fromIntegral remainingEnergy
+        amountWord = _amount amnt
+
+-- |Process a module as received and make a module interface.
+-- This
+-- - checks the module is well-formed, and has the right imports and exports for a V1 module.
+-- - makes a module artifact and allocates it on the Rust side, returning a pointer and a finalizer.
+{-# NOINLINE processModule #-}
```

Comment:
Because of unsafePerformIO, it is there in the V0 version as well. But I don't remember whether there is a reason why I added it to this, but not to the other methods.